tests/unit/backend/corpora/common/entities/test_project.py

```diff
@@ -118,6 +120,52 @@ def test__create_ids__ok(self):
         self.assertEqual(link_ids, get_ids(project.links))
         self.assertNotEqual(["test_project_link_id"], get_ids(project.links))
 
+    def test__list_in_time_range__ok(self):
+        generate = 5
+        sleep = 0.5
+        from_ids = []
+        to_ids = []
+
+        from_date = datetime.now().timestamp()
+        time.sleep(sleep)
+        for i in range(generate):
+            from_ids.append(Project.create(**ProjectParams.get()).id)
+
+        with self.subTest("Test from_date"):
+            # Projects from_date are returned.
+            projects = Project.list_in_time_range(from_date=from_date)
+            for project in projects:
+                self.assertGreaterEqual(project["created_at"].timestamp(), from_date)
+
+        to_date = datetime.now().timestamp()
+        time.sleep(sleep)
+        for i in range(generate):
+            to_ids.append(Project.create(**ProjectParams.get()).id)
+
+        with self.subTest("Test to_date and from_date"):
+            # Projects between to_date and from_date are returned.
+            projects = Project.list_in_time_range(to_date=to_date, from_date=from_date)
+            for project in projects:
+                self.assertGreaterEqual(project["created_at"].timestamp(), from_date)
+                self.assertLessEqual(project["created_at"].timestamp(), to_date)
+
+        with self.subTest("Test to_date"):
+            # All created projects are returned.
+            projects = Project.list_in_time_range(to_date=to_date)
+            self.assertGreater(len(projects), generate)
+            for project in projects:
+                self.assertLessEqual(project["created_at"].timestamp(), to_date)
+
+        with self.subTest("Test no date"):
+            projects = Project.list_in_time_range()
+            test_ids = set([*to_ids, *from_ids])
+            project_ids = set([p["id"] for p in projects])
+            self.assertTrue(test_ids.issubset(project_ids))
+
+        with self.subTest("Verify columns"):
```

Comment:
What exactly is this checking? At the surface-level it seems like you're checking the typing of the returned object but I didn't know if there was a more deeper reasoning for this? It's slightly odd to check typing in a test in Python...

Reply:
Since I don't know the exact value being returned, but I'm certain of the type that should be returned.